pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  callPackage,
+  buildExamples ? false,
+  buildTests ? false,
+  buildAll ? false,
+}:
+
+# Derivation build example:
+#   $ nix-build -A sobjectizer
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    rev = "48240cff83d758f0e7fc413ec408536147a374ed";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = lib.optionals (buildExamples) [ "-DBUILD_EXAMPLES=ON" ] ++
+               lib.optionals (buildTests)    [ "-DBUILD_TESTS=ON" ] ++
+               lib.optionals (buildAll)      [ "-DBUILD_ALL=ON" ];
+
+  doCheck = buildTests || buildAll;
+  checkPhase = ''
+    runHook preCheck
+
+    ctest -C Release --output-on-failure
+
+    runHook postCheck
+  '';
```

Comment:
Does `doCheck = ...` not suffice? Or do you do this just to build the tests in `Release` mode?

Reply:
> Does `doCheck = ...` not suffice? Or do you do this just to build the tests in `Release` mode?

Yes, it does. `doCheck` is enough for tests running. Thanks!